h2o-core/src/main/java/water/H2O.java

```diff
@@ -1550,8 +1554,26 @@ static public void finalizeRegistration() {
   // A dense array indexing all Cloud members. Fast reversal from "member#" to
   // Node.  No holes.  Cloud size is _members.length.
   public final H2ONode[] _memary;
+
+  // mapping from a node ip to node index
+  private final HashMap<String, Integer> _node_ip_to_index;
   final int _hash;
 
+  public H2ONode getNodeByIpPort(String ipPort){
+    if(_node_ip_to_index != null) {
+      Integer index = _node_ip_to_index.get(ipPort);
+      if (index != null && index > -1 && index < _memary.length) {
```

Comment:
Can index ever be >= _memary.lenght?

I think this would indicate a bug, and  it is therefore better to throw an exception than to ignore the inconsistency.